### PR TITLE
use pppmap in grid-search

### PR DESCRIPTION
**Pull request by behrica**
allows to see the progress of the gridsearch as logging, if wanted

### CODE REVIEW

File: src/tech/v3/ml.clj [hunk 1]
```diff
@@ -249,7 +251,7 @@ see tech.v3.dataset.modelling/set-inference-target")
              (log/warn "Did not find any gridsearch axis in options map"))
          ds-seq (ds-mod/k-fold-datasets dataset n-k-folds gridsearch-options)]
      (->> gs-seq
-          (pmap #(do-k-fold % loss-fn target-colname ds-seq))
+          (ppp/ppmap-with-progress "gridsearch" 1 #(do-k-fold % loss-fn target-colname ds-seq))
```
Comment (behrica) [inline src/tech/v3/ml.clj]: To set the "grain size" smartly might improve the performance.

As you are aware, `pmap` is not very efficient, if the tasks are very short (and we have a lot of tasks), as then there is more overhead in multi-core then gained by the parallel working.

The "best" grain size depends therefore on the size/duration of the tasks, which is unknown.

The grain size of 1 is the same as "pmap"

Comment (cnuernber) [reply src/tech/v3/ml.clj]: I think in general k-fold is expensive enough that a grain size of 1 will activate full parallelism.

**Comment by cnuernber** [comment]
Just a few conflicts due to merging other PRs.

File: src/tech/v3/ml.clj (hunk 1 above)
Comment (behrica) [reply src/tech/v3/ml.clj]: yes, indeed , the task here is a "k-fold".

so we would only gain the progress report by this PR.

I would like to have it in, in any case.

I call the long running code by piping to a logfile, and can then run "tail -f log.txt | grep "gridsearch".
to get a high level progress on the level of the grid searches.

Something similar could be done by re-configure the logging to a write to a file and grep on that.

**Comment by behrica** [comment]
Please wait, I i messed the rebase...

**Comment by behrica** [comment]
No, was fine.

**Pull request merged**